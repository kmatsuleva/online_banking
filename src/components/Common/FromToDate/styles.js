const styles = {
    container: {
        display: "flex",
        alignContent: "flex-start",
        alignItems: "baseline",
    }
}

export default styles;