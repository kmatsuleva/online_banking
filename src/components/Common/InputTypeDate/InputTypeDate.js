import React from "react";
import PropTypes from 'prop-types';     

const InputTypeDate = ({  min, max, style }) => (
    <input type="date" min={min} max = {max} style={style}/>  
);


InputTypeDate.propTypes = {
    min: PropTypes.string,
    max: PropTypes.string,
    style: PropTypes.object  
};

InputTypeDate.defaultProps = {
    style: {
        border: "1px solid #d2d2d2",
        padding: "8px 10px",
        marginTop: 8,
        marginBottom: 15,
        borderRadius: 3,
        height: 23,
        fontSize: 15,
        color: "#4a4a4a",
        boxShadow: "inset 1px 1px 2px rgba(0, 0, 0, 0.2)",
        width: 170,
        fontFamily: "Roboto, Arial, sans-serif"
    },

    min: "2018-01-01",
    max: "2022-01-01"
}



export default InputTypeDate;


