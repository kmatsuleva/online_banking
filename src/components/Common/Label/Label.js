import React from "react";
import PropTypes from 'prop-types';     

const Label = ({ text, className }) => (
  <label className= {className}> { text } </label>
);

Label.propTypes = {
  text: PropTypes.string,
  style: PropTypes.object    
};



export default Label;
