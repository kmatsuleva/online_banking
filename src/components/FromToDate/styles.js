const styles = {
    container: {
        display: "flex",
        alignContent: "flex-start",
        alignItems: "baseline",
    },

    label: {
        padding: "0 30px"
    }
}

export default styles;