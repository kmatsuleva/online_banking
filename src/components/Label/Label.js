import React from "react";
import styles from "./styles.js"

const Label = ({ labelValue }) => (
  <div>
      <label style={styles.label} > {labelValue} </label>
  </div>
);

export default Label;
 