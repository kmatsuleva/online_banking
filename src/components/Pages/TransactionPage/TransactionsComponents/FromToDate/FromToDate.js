import React from "react";
import styles from "./styles"

const FromToDate = () => (
        <form style={styles.container}>
                <label style={styles.label}>From: </label>
                <input type="date" min="2013-12-31" style={styles.inputDate}/>
                <label style={styles.label}>To: </label>
                <input type="date" min="2013-12-31" style={styles.inputDate}/>
        </form>
);

export default FromToDate;     


