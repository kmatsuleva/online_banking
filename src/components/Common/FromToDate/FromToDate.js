import React from "react";
import Label from "../Label/Label"  
import InputTypeDate from "../InputTypeDate/InputTypeDate" 


const FromToDate = () => (
        <form >
                <Label text = "From:" style={labelStyles}/>
                <InputTypeDate min="2018-01-01" max="2022-08-09"/>
                <Label text ="To: " style={labelStyles}/>
                <InputTypeDate min="2018-01-01" max="2022-08-09"/>
        </form>
);

const labelStyles = {
        fontSize: 18,
        color: "#15517d",
        fontFamily: "Roboto, Arial, sans-serif",
        marginBottom: 8,
        padding: "0 30px"
};


export default FromToDate;     


