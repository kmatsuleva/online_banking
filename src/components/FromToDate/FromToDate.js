import React from "react";
import Label from "../Label/Label";
import InputDate from '../InputTypeDate/InputDate'
import styles from "./styles"

const FromToDate = () => (
        <form style={styles.container}>
                <div style={styles.label}><Label labelValue = "From:"/></div>
                <InputDate />
                <div style={styles.label}><Label labelValue = "To:"/></div>
                <InputDate />
        </form>
);

export default FromToDate;     


