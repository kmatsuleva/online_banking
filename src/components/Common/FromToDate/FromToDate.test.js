
import React from 'react'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import FromToDate from "./FromToDate"
import Label from "../Label/Label"  
import InputTypeDate from "../InputTypeDate/InputTypeDate"  

Enzyme.configure({ adapter: new Adapter() })

describe("< FromToDate />", () => {

    it('renders without crashing', () => {
        shallow(<FromToDate />);
    });

    it('should render correctly', () => {
        const output = shallow(
        <FromToDate  />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it("renders <Label /> component", () => {
        const wrapper = shallow(<FromToDate />);
        expect(wrapper.find(Label).length).toEqual(2);
    });

    it("renders <InputTypeDate /> component", () => {
        const wrapper = shallow(<FromToDate />);
        expect(wrapper.find(InputTypeDate).length).toEqual(2);
    });
});