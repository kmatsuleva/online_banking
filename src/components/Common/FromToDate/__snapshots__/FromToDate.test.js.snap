// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< FromToDate /> should render correctly 1`] = `
<form>
  <Label
    style={
      Object {
        "color": "#15517d",
        "fontFamily": "Roboto, Arial, sans-serif",
        "fontSize": 18,
        "marginBottom": 8,
        "padding": "0 30px",
      }
    }
    text="From:"
  />
  <InputTypeDate
    max="2022-08-09"
    min="2018-01-01"
    style={
      Object {
        "border": "1px solid #d2d2d2",
        "borderRadius": 3,
        "boxShadow": "inset 1px 1px 2px rgba(0, 0, 0, 0.2)",
        "color": "#4a4a4a",
        "fontFamily": "Roboto, Arial, sans-serif",
        "fontSize": 15,
        "height": 23,
        "marginBottom": 15,
        "marginTop": 8,
        "padding": "8px 10px",
        "width": 170,
      }
    }
  />
  <Label
    style={
      Object {
        "color": "#15517d",
        "fontFamily": "Roboto, Arial, sans-serif",
        "fontSize": 18,
        "marginBottom": 8,
        "padding": "0 30px",
      }
    }
    text="To: "
  />
  <InputTypeDate
    max="2022-08-09"
    min="2018-01-01"
    style={
      Object {
        "border": "1px solid #d2d2d2",
        "borderRadius": 3,
        "boxShadow": "inset 1px 1px 2px rgba(0, 0, 0, 0.2)",
        "color": "#4a4a4a",
        "fontFamily": "Roboto, Arial, sans-serif",
        "fontSize": 15,
        "height": 23,
        "marginBottom": 15,
        "marginTop": 8,
        "padding": "8px 10px",
        "width": 170,
      }
    }
  />
</form>
`;
